graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphqlServiceBuilder.java

```diff
@@ -145,19 +152,42 @@ public GraphqlServiceBuilder schema(GraphQLSchema schema) {
         return this;
     }
 
+    /**
+     * Sets {@link DataLoaderRegistry} creation function.
+     */
+    public GraphqlServiceBuilder dataLoaderRegistry(
+            Function<? super ServiceRequestContext, ? extends DataLoaderRegistry> dataLoaderRegistryFactory) {
+        checkArgument(dataLoaderRegistryConsumers == null,
+                      "configureDataLoaderRegistry() and dataLoaderRegistry() are mutually exclusive.");
```

Comment:
`IllegalStateException` is a proper exception because it checks `dataLoaderRegistryConsumers` status and does not validate the input.
```suggestion
        checkState(dataLoaderRegistryConsumers == null,
                   "configureDataLoaderRegistry() and dataLoaderRegistry() are mutually exclusive.");
```

Reply:
Good point! 👍